presto-geospatial/src/main/java/com/facebook/presto/plugin/geospatial/GeoFunctions.java

```diff
@@ -381,6 +383,43 @@ public static Long stNumInteriorRings(@SqlType(GEOMETRY_TYPE_NAME) Slice input)
         return Long.valueOf(((OGCPolygon) geometry).numInteriorRing());
     }
 
+    @Description("Returns the cardinality of the geometry collection")
+    @ScalarFunction("ST_NumGeometries")
+    @SqlType(StandardTypes.INTEGER)
+    public static long stNumGeometries(@SqlType(GEOMETRY_TYPE_NAME) Slice input)
+    {
+        OGCGeometry geometry = deserialize(input);
+        if (geometry.isEmpty()) {
+            return 0;
+        }
+        GeometryType type = GeometryType.getForEsriGeometryType(geometry.geometryType());
+        if (!type.isMultitype()) {
+            return 1;
+        }
+        return ((OGCGeometryCollection) geometry).numGeometries();
+    }
+
+    @SqlNullable
+    @Description("Returns the geometry element at the specified index, index started with 1")
+    @ScalarFunction("ST_GeometryN")
+    @SqlType(GEOMETRY_TYPE_NAME)
+    public static Slice stGeometryN(@SqlType(GEOMETRY_TYPE_NAME) Slice input, @SqlType(INTEGER) long index)
+    {
+        OGCGeometry geometry = deserialize(input);
+        GeometryType type = GeometryType.getForEsriGeometryType(geometry.geometryType());
+        if (!type.isMultitype()) {
```

Comment:
@dongqingx When geometry is empty and index is 1 this code seems to return the geometry itself instead of null. Would you double check by adding tests and fix the code if needed?

Reply:
fix it.